perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -0,0 +1,162 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include "sim86_shared.h"
+#pragma comment (lib, "sim86_shared_debug.lib")
+#define ArrayCount(Array) (sizeof(Array) / sizeof((Array)[0]))
+
+s16 Registers[15] = {};
+
+char const* RegisterNames[][3] =
+{
+    {"", "", ""},
+    {"al", "ah", "ax"},
+    {"bl", "bh", "bx"},
+    {"cl", "ch", "cx"},
+    {"dl", "dh", "dx"},
+    {"sp", "sp", "sp"},
+    {"bp", "bp", "bp"},
+    {"si", "si", "si"},
+    {"di", "di", "di"},
+    {"es", "es", "es"},
+    {"cs", "cs", "cs"},
+    {"ss", "ss", "ss"},
+    {"ds", "ds", "ds"},
+    {"ip", "ip", "ip"},
+    {"flags", "flags", "flags"}
+};
+
+void SimulateInstruction(instruction Instruction)
+{
+    const char* Op = Sim86_MnemonicFromOperationType(Instruction.Op);
+    
+    if (Instruction.Op == Op_mov)
+    {
+        instruction_operand Dest = Instruction.Operands[0];
+        instruction_operand Source = Instruction.Operands[1];
+        s32 Value; // s32 to avoid issues on the bitshift right
+        switch (Source.Type)
+        {
+            case Operand_Register:
+            {
+                Value = Registers[Source.Register.Index];
+                if (Source.Register.Offset == 1) // Accessing high 8 bits
+                {
+                    Value >>= 8;
+                }
+                else if (Source.Register.Count == 1) // Accessing low 8 bits
+                {
+                    Value &= 0xFF;
+                }
```

Comment:
This code could be made clearer. Instead of testing one thing, and then saying something else in the comments, you could just test what you really want directly.

For example, accessing high bits could be
```
if (Source.Register.Count == 1)
{
    if (Source.Register.Offset == 1)
    {
        ...
    }
}
```

which is a lot more explicit.

This is just an example, there are other ways to write this, for example, maybe I would do something like this instead:

```
Value >>= 8 * Source.Register.Offset;
```

Reply:
I liked yours better.
Fixed
e538a361007225c0afc7d0540ba7e3896af95d51